ansible/roles/test/tasks/continuous_link_flap.yml

```diff
@@ -0,0 +1,196 @@
+# This is continuous link flap test. In this test,
+# 1.) Flap all interfaces one by one in 1-3 iteration
+# to cause BGP Flaps.
+# 2.) Flap all interfaces on peer (FanOutLeaf) one by one 1-3 iteration
+# to cause BGP Flaps.
+# 3.) Watch for memory (show system-memory) ,orchagent CPU Utilization
+# and Redis_memory.
+# Pass Criteria: All routes must be re-learned with < 5% increase in Redis
+# and ORCH agent CPU consumption below 10% after 3 mins after stopping flaps.
+
+- block:
+    - name: Record  memory status at start
+      shell: show system-memory
+      register: memory_output
+
+    - debug: msg="Memory Status {{memory_output.stdout}}"
+
+    - name: Record Redis Memory at start
+      shell: redis-cli info memory | grep used_memory_human | sed -e 's/.*:\(.*\)M/\1/'
+      register: start_time_redis_memory
+
+    - debug: msg="Redis Memory {{start_time_redis_memory.stdout}} M"
+
+# track only ipv4 routes. IPv6 addresses are removed with ifconfig
+# down command, which may bring ipv6 BGP neighborship down. This
+# must be addressed in dev code first.
+    - name: Record ipv4 route counts at start
+      shell: show ip route summary | grep Total | awk '{print $2}'
+      register: start_time_ipv4_route_counts
+
+    - debug: msg="IPv4 routes at start {{start_time_ipv4_route_counts.stdout}}"
+
+# Make Sure Orch CPU < 10 before starting test.
+    - name: Make Sure orchagent CPU utilization is less that 10 before link flap
+      shell: show processes cpu | grep orchagent | awk '{print $9}'
+      register: orch_cpu
+      until: "{{orch_cpu.stdout|int}} < 10"
+      retries: 100
+      delay: 2
+
+    - debug: msg="First Iteration flap all interfaces one by one on DUT"
+
+    - name: Gathering lab graph facts about the device
+      conn_graph_facts: host={{ inventory_hostname }}
+      connection: local
+      tags: always
+
+    - debug: msg="{{ device_conn }}"
+
+    - name: include continuous_link_flap_helper.yml
+      include: continuous_link_flap/continuous_link_flap_helper.yml interface="{{item}}"
+      with_items: "{{ device_conn.keys() }}"
+
+    - debug: msg="Second Iteration flap all interfaces one by one on DUT"
+
+    - name: Gathering lab graph facts about the device
+      conn_graph_facts: host={{ inventory_hostname }}
+      connection: local
+      tags: always
+
+    - name: include continuous_link_flap_helper.yml
+      include: continuous_link_flap/continuous_link_flap_helper.yml interface="{{item}}"
+      with_items: "{{ device_conn.keys() }}"
+
+    - debug: msg="Third Iteration flap all interfaces one by one DUT"
+
+    - name: Gathering lab graph facts about the device
+      conn_graph_facts: host={{ inventory_hostname }}
+      connection: local
+      tags: always
```

Comment:
why not use a loop and specify the loop count with a parameter.

Reply:
As I know, Ansible does not support loop over a block or a double loop. And I do not want to write the block in another file and use include. So I thought lets have in plain way in same file, it is better for readability too. 
Let me know If you have suggestion on this,  I can learn new looping in Ansible. Thx.

Reply:
Let`s mark it as fixed as of now.

Reply:
Did you try `with_items: [1, 2, 3]` ?

---
In reply to: [274190641](https://github.com/Azure/sonic-mgmt/pull/810#discussion_r274190641) [](ancestors = 274190641)

Reply:
Still asking.

---
In reply to: [278333252](https://github.com/Azure/sonic-mgmt/pull/810#discussion_r278333252) [](ancestors = 278333252,274190641)

Reply:
Do we have an example in sonic_mgmt repo where loop in loop works with 'with_items' ? 
'with_nested' works to create a double loop over a single ansible module, but looping over a block is still not supported.

Reply:
For example, below will not print msg 9 times.
  tasks:
   - debug: msg="{{item}}"
     with_items: [1, 2, 3]
     with_items: [4, 5, 6]

I will run only 3 times as below:
TASK [debug] *******************************************************************
Thursday 25 April 2019  22:45:34 +0000 (0:00:01.144)       0:00:01.167 ********
ok: [falco-test-dut01] => (item=4) => {
    "item": 4,
    "msg": 4
}
ok: [falco-test-dut01] => (item=5) => {
    "item": 5,
    "msg": 5
}
ok: [falco-test-dut01] => (item=6) => {
    "item": 6,
    "msg": 6
}